pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -286,11 +288,18 @@ func (configgen *ConfigGeneratorImpl) buildGatewayHTTPRouteConfig(env *model.Env
 	}
 	// call plugins
 	for _, p := range configgen.Plugins {
+		var perRouteFilterConfig map[string]*types.Struct
+		if c, ok := configgen.PerRouteFilterConfig[p.GetName()].(map[string]*types.Struct); ok {
+			perRouteFilterConfig = c
+		} else {
+			perRouteFilterConfig = nil
+		}
 		in := &plugin.InputParams{
-			ListenerProtocol: plugin.ListenerProtocolHTTP,
-			Env:              env,
-			Node:             node,
-			Push:             push,
+			ListenerProtocol:     plugin.ListenerProtocolHTTP,
+			Env:                  env,
+			Node:                 node,
+			Push:                 push,
+			PerRouteFilterConfig: perRouteFilterConfig,
```

Comment:
why dont you just pass the SidecarScope here? And let the plugins query the sidecarScope to see if they have cached anything there.